### PR TITLE
Fix several broken features.

### PR DESCRIPTION
**Pull request by sroj**
The main issue being dealt with here is that the RPC result message handling was broken.

The problem was in the `deserialize` method: the result messages ended up falling into the wrong conditional branch; see https://github.com/OzTK/DDPClient.NET/pull/6#discussion_r48902662.

Other issues fixed are detailed here (copied from 429207b's message):
- Handle RPC result messages properly.
- Handle RPC updated messages properly.
- Handle connected and failed messages properly.
- Publish error messages to message subscribers.
- Parse previously non parsed fields in error messages.
- Handle non error messages in a consistent fashion.
- Uppercase first letter only, instead of doing TitleCase. This
  produces conforming variable names.
- Cast a dynamic type to an IDictionary. Just to make this client
  compatible with Xamarin.iOS. This should not produce any difference
  in behavior in other .NET platforms.

### CODE REVIEW

File: Net.DDP.Client/JsonDeserializeHelper.cs [hunk 1]
```diff
@@ -3,141 +3,147 @@
...
-
-        internal void Deserialize(string jsonItem)
-        {
-            JObject jObj = JObject.Parse(jsonItem);
-
-            if (jObj[DDPClient.DDP_PROPS_ERROR] != null || jObj[DDPClient.DDP_PROPS_MESSAGE] != null && jObj[DDPClient.DDP_PROPS_MESSAGE].ToString() == "error")
-                HandleError(jObj[DDPClient.DDP_PROPS_ERROR] ?? jObj);
-            else if (jObj[DDPClient.DDP_PROPS_SESSION] != null)
-                HandleConnected(jObj);
-            else if (jObj[DDPClient.DDP_PROPS_MESSAGE] != null)
```
Comment (sroj) [inline Net.DDP.Client/JsonDeserializeHelper.cs:26]: This was the main offending line: a result message _does_ contain a message field (every DDP message does), so controls incorrectly falls into this branch of the conditional, and therefore the parsing of the result message never happens.

**Comment by OzTK** [comment]
Thank you for the contribution to align to the spec. I will test it and merge when I have some more time as well as publish an update to the nuget package. Do you work only with Xamarin on a non-windows environment?

**Comment by OzTK** [comment]
Thank you for the contribution to align to the spec. I will test it and merge when I have some more time as well as publish an update to the nuget package. Do you work only with Xamarin on a non-windows environment?

**Comment by sroj** [comment]
Cool, sounds good. In the context of this pull request and project, yes, only with Xamarin/OS X. In general, no.

**Comment by sroj** [comment]
Closing due to lack of activity

**Pull request closed without merging**